src/Compiler/Compiler.php

```diff
@@ -182,9 +182,12 @@ private function compileDefinition(string $entryName, Definition $definition) :
                 }
                 break;
             case $definition instanceof StringDefinition:
-                $entryName = $this->compileValue($definition->getName());
-                $expression = $this->compileValue($definition->getExpression());
-                $code = 'return \DI\Definition\StringDefinition::resolveExpression(' . $entryName . ', ' . $expression . ', $this->delegateContainer);';
+                $expression = $definition->getExpression();
+                $callback = function (array $matches) use ($definition) {
+                    return '\'.$this->resolveStringPlaceholder(' . $this->compileValue($matches[1]) . ', ' . $this->compileValue($definition->getName()) . ').\'';
```

Comment:
I would use `sprintf` instead of dots concatenation

Reply:
missing test cases

Reply:
@partikus according to my perf test:
```
<?php
define('NUM_TESTS', 1000000);
header('Content-type: text/plain');

echo NUM_TESTS . " iterations:\n";

$a = 'test';
$b = 'abc';
$c = 'gggggggg';

$start = microtime(true);
$instance = 0;
for ($i=0; $i<NUM_TESTS; $i++) {
	$ref = $a . $b . 'blablabla' . $c;
}
$end = microtime(true);
$result1 = $end - $start;
echo '#1 : '. number_format($result1 * 1000, 3) . " ms\n";

$start = microtime(true);
$instance = 0;
for ($i=0; $i<NUM_TESTS; $i++) {
	$ref = sprintf('%s%sblablabla%s', $a, $b, $c);
}
$end = microtime(true);
$result2 = $end - $start;
echo '#2 : '. number_format($result2 * 1000, 3) . " ms\n";

echo 'diff: ' . ($result2 / $result1) * 100 . '%';
```

sprintf might be ~5 times slower than concatenation:
```
1000000 iterations:
#1 : 217.159 ms
#2 : 1,088.653 ms
diff: 501.31594115703%
```

The compiled container is targeted for maximum performance, code readability is not important there.

Reply:
@falkenhawk thanks for the test case, I have ran the above example on my local machine and the difference was between 130-170%
but still it's valuable change, 
lets stay with dots :+1: